site/content/en/docs/Getting started/first_kustomization.md

```diff
@@ -426,3 +426,396 @@ spec:
+kind: Deployment
+metadata:
+  labels:
+    app: nginx
+    variant: var1                ### label has been set here
+  name: env1-example-nginx-production
+  namespace: production
+spec:
+  replicas: 3
+  selector:
+    matchLabels:
+      app: nginx
+  template:
+    metadata:
+      labels:
+        app: nginx
+        variant: var1            ### label has been set here
+    spec:
+      containers:
+      - image: nginx:1.20.2      ### image tag has been set to 1.20.2
+        name: nginx
+        ports:
+        - containerPort: 80
+```
+
+The last requirement to meet is to set the environment variable, and to do that you will create a patch.  To do this, create the following file for the `Production` overlay:
+
+```bash
+cat <<'EOF' >overlays/production/patch-env-vars.yaml
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: example-nginx
+spec:
+  template:
+    spec:
+      containers:
+      - name: nginx
+        env:
+        - name: ENVIRONMENT
+          value: prod
+EOF
+```
+
+Next step, add a reference to that patch file in `kustomization.yaml`:
+
+```yaml
+patches:
+- patch-env-vars.yaml
+```
+
+One important thing to note here is that the name of the deployment used is the name that you are getting from the base and not the deployment name that has the prefix and suffix added.
+
+Rebuilding the overlay shows that the environment variable has been added to your container:
+
+```yaml
+$ kustomize build overlays/production/
+
+apiVersion: v1
+kind: Service
+metadata:
+  labels:
+    app: nginx
+    variant: var1
+  name: env1-example-nginx-production
+  namespace: production
+spec:
+  ports:
+  - port: 80
+    protocol: TCP
+    targetPort: 80
+  selector:
+    app: nginx
+---
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  labels:
+    app: nginx
+    variant: var1
+  name: env1-example-nginx-production
+  namespace: production
+spec:
+  replicas: 3
+  selector:
+    matchLabels:
+      app: nginx
+  template:
+    metadata:
+      labels:
+        app: nginx
+        variant: var1
+    spec:
+      containers:
+      - env:
+        - name: ENVIRONMENT               ### Environment variable has been added here
+          value: prod
+        image: nginx:1.20.2
+        name: nginx
+        ports:
+        - containerPort: 80
+```
+
+Looking at the output of `kustomize build` you can see that the additional requirements that were set have now been met.  Below are the files as they should be at this point in your overlays and the `kustomize build` output:
+
+_kustomize-example/overlays/production/kustomization.yaml_:
+```yaml
+apiVersion: kustomize.config.k8s.io/v1beta1
+kind: Kustomization
+
+namePrefix: env1-
+nameSuffix: -production
+
+namespace: production
+
+replicas:
+- name: example-nginx
+  count: 3
+
+images:
+- name: nginx
+  newTag: 1.20.2
+
+labels:
+- pairs:
+    variant: var1
+  includeSelectors: false
+  includeTemplates: true
+
+resources:
+- ../../base
+
+patches:
+- patch-env-vars.yaml
```

Comment:
```suggestion
- path: patch-env-vars.yaml
```

Reply:
this works for me without `path:` as well, is that intended?

Updated all occurrences in this page as suggested

Reply:
I didn't even know it myself until recently, but it seems that a long time ago, the `patches` field was essentially the `patchesStrategicMerge` field, and Kustomize still detects the now-incorrect syntax and silently converts it internally to `patchesStrategicMerge`, making paths and raw patches work (as long as they contain SMPs).

Reply:
thanks for the information!